src/queries/rag.ts

```diff
@@ -0,0 +1,60 @@
+import type { DataEmbedderObject, Model, Prompt, Retriever } from '../types';
+
+export type RAGOptions = {
+  model: Model;
+  prompt: Prompt;
+  retriever: Retriever;
+  embedder: DataEmbedderObject;
+};
+
+export class RAG {
+  private retriever: Retriever;
+  private model: Model;
+  private prompt: Prompt;
+  private embedder: DataEmbedderObject;
+
+  constructor(options: RAGOptions) {
+    this.model = options.model;
+    this.prompt = options.prompt;
+    this.retriever = options.retriever;
+    this.embedder = options.embedder;
+  }
+
+  async run(query: string) {
+    const embeddings = await this.embedder.embed(query);
+
+    const context = await this.retriever.retrieve(embeddings[0]);
```

Comment:
Do we know that we will only ever have one? It makes sense to do only `[0]` to start, but maybe worth a comment if we're going to add that later.